### PR TITLE
[Chore] Unit tests for SecureStore

### LINKED ISSUE #411: [Tests] Unit tests for SecureStore

### PR DESCRIPTION
**Pull request by inf2381**
Fixes #411

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+	func testResultReceivedTimeStamp_Success() {
+		XCTAssertNil(store.testResultReceivedTimeStamp)
+		store.testResultReceivedTimeStamp = Int64.max
+		XCTAssertEqual(store.testResultReceivedTimeStamp, Int64.max)
+		store.testResultReceivedTimeStamp = Int64.min
+		XCTAssertEqual(store.testResultReceivedTimeStamp, Int64.min)
+	}
+
+	func lastSuccessfulSubmitDiagnosisKeyTimestamp_Success() {
```
Comment (llanghammer-sap) [inline src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift:38]: Xcode doesn't seem to recognize functions in Unit Tests without a prefix of 'test'. This applies to all following functions. 

If this was on purpose feel free to resolve this conversation

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		XCTAssertNil(store.lastSuccessfulSubmitDiagnosisKeyTimestamp)
+		store.lastSuccessfulSubmitDiagnosisKeyTimestamp = Int64.max
+		XCTAssertEqual(store.lastSuccessfulSubmitDiagnosisKeyTimestamp, Int64.max)
+		store.lastSuccessfulSubmitDiagnosisKeyTimestamp = Int64.min
+		XCTAssertEqual(store.lastSuccessfulSubmitDiagnosisKeyTimestamp, Int64.min)
+	}
+
+	func numberOfSuccesfulSubmissions_Success() {
+		XCTAssertNil(store.numberOfSuccesfulSubmissions)
```
Comment (llanghammer-sap) [inline src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift:47]: Currently returns 0 not nil.

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		store.numberOfSuccesfulSubmissions = Int64.max
+		XCTAssertEqual(store.numberOfSuccesfulSubmissions, Int64.max)
+		store.numberOfSuccesfulSubmissions = Int64.min
+		XCTAssertEqual(store.numberOfSuccesfulSubmissions, Int64.min)
+	}
+
+	func initialSubmitCompleted_Success() {
+		XCTAssertNil(store.initialSubmitCompleted)
```
Comment (llanghammer-sap) [inline src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift:55]: Currently returns _false_ and not _nil_

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		store.initialSubmitCompleted = true
+		XCTAssertEqual(store.initialSubmitCompleted, true)
+		store.initialSubmitCompleted = false
+		XCTAssertEqual(store.initialSubmitCompleted, false)
+	}
+
+	func registrationToken_Success() {
+		XCTAssertEqual(store.registrationToken, "")
```
Comment (llanghammer-sap) [inline src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift:63]: Currently returns _nil_ and not _""_

**Review by llanghammer-sap — CHANGES_REQUESTED** [review]
Currently only the first test seems to run, as Xcode seems to exspect a prefix of "test" for all functions. I don't think that was on purpose.

There are also so failing test, where we maybe should have a look at the exspected behaviour of the store. I marked all of them with the value they are currently returning.

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift (hunk 2 above)
Comment (inf2381) [reply src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift]: Fixed

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift (hunk 1 above)
Comment (inf2381) [reply src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift]: Fixed

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift (hunk 3 above)
Comment (inf2381) [reply src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift]: Fixed

File: src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift (hunk 4 above)
Comment (inf2381) [reply src/xcode/ENA/ENA/Source/Workers/__tests__/StoreTests.swift]: Fixed

**Review by ChristianKienle — APPROVED** [review]
Super cool tests. We should add a few more tests though. Especially the following should also be covered:

We store the summary + the tracing history inside the store. This means that we can only ever make `Codable`-compatible changes to those structs. Thus we should use a store file - add it to our tests as a fixture and then try to modify it in our tests using the current coding. This test ensures that old "data" can still be read/written using new code.

**Review by llanghammer-sap — APPROVED** [review]

**Review by ChristianKienle — CHANGES_REQUESTED** [review]
whops

**Review by ChristianKienle — APPROVED** [review]

**Pull request merged**